tests/jsunit/xml_test.js

```diff
@@ -75,8 +77,35 @@ function test_domToWorkspace() {
         '  </block>' +
         '</xml>');
     var workspace = new Blockly.Workspace();
-    Blockly.Xml.domToWorkspace(dom, workspace);
+	Blockly.Xml.domToWorkspace(dom, workspace);
     assertEquals('Block count', 1, workspace.getAllBlocks().length);
+	Blockly.Xml.domToWorkspace(dom, workspace,[50,50]);
+	assertEquals('Block count', 2, workspace.getAllBlocks().length);
+  } finally {
+    delete Blockly.Blocks.test_block;
+  }
+}
+
+function test_appendDomToWorkspace() {
+  Blockly.Blocks.test_block = {
+    init: function() {
+      this.jsonInit({
+        message0: 'test',
+      });
+    }
+  };
+
+  try {
+    var dom = Blockly.Xml.textToDom(
+        '<xml xmlns="http://www.w3.org/1999/xhtml">' +
+        '  <block type="test_block" inline="true" x="21" y="23">' +
+        '  </block>' +
+        '</xml>');
+    var workspaceA = new Blockly.Workspace();
```

Comment:
I try to directly create a workspaceSvg
var workspaceB = new Blockly.WorkspaceSvg({toolbox: XML_TOOLBOX});
where var XML_TOOLBOX = ['<xml id="toolbox" style="display: none">',
 '  <category name="Logic" colour="210">',
 '     <block type="controls_if"></block>',
 '     <block type="logic_compare"></block>',
 '     <block type="logic_operation"></block>',
 '     <block type="logic_negate"></block>',
 '     <block type="logic_boolean"></block>',
 '     <block type="logic_null"></block>',
 '     <block type="logic_ternary"></block>',
 '   </category>',
 '</xml>'].join('\n');  

Then I inject a block from an xml snippet
Blockly.Xml.domToWorkspace(dom, workspaceB);
But I get 
this.workspace.getCanvas(...) is undefined
> Blockly.BlockSvg.prototype.initSvg at file:///blockly/core/block_svg.js:147:5
> Blockly.Xml.domToBlock at file:///blockly/core/xml.js:435:9
> Blockly.Xml.domToWorkspace

I guess the way I call the workspaceSvg is too simple but I tried to avoid the html version of it Blockly.Inject().